experimental/dds/tree2/src/shared-tree-core/editManagerFormat.ts

```diff
@@ -0,0 +1,63 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TSchema, Type } from "@sinclair/typebox";
+import { Brand, brandedNumberType } from "../util";
+import { SessionId, SessionIdSchema, RevisionTag, RevisionTagSchema } from "../core";
+
+/**
+ * Contains a single change to the `SharedTree` and associated metadata.
+ */
+export interface Commit<TChangeset> {
+	readonly revision: RevisionTag;
+	readonly change: TChangeset;
+	/** An identifier representing the session/user/client that made this commit */
+	readonly sessionId: SessionId;
+}
+
+const Commit = <ChangeSchema extends TSchema>(tChange: ChangeSchema) =>
+	Type.Object({
+		revision: RevisionTagSchema,
+		change: tChange,
+		sessionId: SessionIdSchema,
+		// Enforce that `parent` isn't present. Commits are generally encoded from `GraphCommit`s, which can
```

Comment:
I'm slightly concerned about the discoverability of this when making changes to GraphCommits. The commit type is also not super clear as an encoding type (which I'm starting to think it is from this PR but again, not super clear). I wonder if it's reasonable to more explicitly force conversion to a Commit before encoding, if it's not used for anything else.

Reply:
see guidelines on #14954 (would love to check that in, it's in a complete-enough state that I've been acting on it). specifically, any file ending in `Format` contains persisted types.

w.r.t encoding woes, I refactored this to use `{ additionalProps: false }`, which I'd rather do if there's more than one field omitted anyway--I had considered that originally and was on the fence before on which to use.